src/libs/ReportUtils.js

```diff
@@ -1874,44 +1897,54 @@ function isUnreadWithMention(report) {
 }
 
 /**
- * Determines if a report has an outstanding IOU that doesn't belong to the currently logged in user
+ * Determines if a report has an outstanding IOU that is waiting for an action from the current user
  *
  * @param {Object} report
  * @param {String} report.iouReportID
- * @param {Object} iouReports
  * @returns {boolean}
  */
-function hasOutstandingIOU(report, iouReports) {
-    if (!report || !report.iouReportID || _.isUndefined(report.hasOutstandingIOU)) {
-        return false;
+function isWaitingForIOUActionFromCurrentUser(report) {
+    let reportToLook = report;
+
+    if (report.iouReportID) {
+        const iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${report.iouReportID}`];
+        if (!iouReport) {
+            return false;
+        }
+
+        reportToLook = iouReport;
     }
 
-    const iouReport = iouReports && iouReports[`${ONYXKEYS.COLLECTION.REPORT}${report.iouReportID}`];
-    if (!iouReport || !iouReport.ownerAccountID) {
+    if (!reportToLook.ownerAccountID) {
         return false;
     }
 
-    if (iouReport.ownerAccountID === currentUserAccountID) {
-        return false;
+    if (reportToLook.ownerAccountID === currentUserAccountID && reportToLook.isWaitingOnBankAccount) {
+        return true;
     }
 
-    return report.hasOutstandingIOU;
+    if (reportToLook.ownerAccountID !== currentUserAccountID && reportToLook.hasOutstandingIOU) {
```

Comment:
hmm I don't see why?

Reply:
We've been using `hasOutstandingIOU` as a way to tell if the report is settled or not, but we have report states for that. It'd be nice if we could standardize in one key for it and eventually remove the other.

Reply:
but now, `hasOutstandingIOU` doesn't necesserily mean `!isSettled`. Because when `isWaitingForBankAccount`, both are actually false. But I see your point. I'll try to think of a way to uniformize everything.